### ISSUE TITLE
[Bug]: trace/log's merge cron_task cost 6 GB Memory

### ISSUE THREAD

**Issue by xzxiong** [matrixorigin/matrixone#8354]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

the following chart show that: objectio.newDecompressToObject alloc **6756.07** MB
<img width="1138" alt="企业微信截图_ada10706-79d3-4ea0-aec6-2cc75bd525c8" src="https://user-images.githubusercontent.com/3927687/224214691-62e48e0c-524b-4231-9d03-e2ff33838bcd.png">

**Comment by xzxiong** [comment]
the memory alloc is here.

apply same size for compressed data.

It will cause this issue situation, but the compressed data is just  1/10 ~ 1/30 of original data.

![image](https://user-images.githubusercontent.com/3927687/224236388-a884f345-77db-42ee-8e61-cd5d9e29ef63.png)

**Comment by xzxiong** [comment]
here is the memory cost, after with debug code : reallocCompressedData, 

then memory max usage <= 128MB * 2. (128 MB is the merged target filesize)

![image](https://user-images.githubusercontent.com/3927687/224237949-de740cb5-426b-4ed7-a59a-66a3ef70451a.png)

**Comment by xzxiong** [comment]
Write issue will be fixed in https://github.com/matrixorigin/matrixone/issues/8368
Read issue still on analysising

**Comment by xzxiong** [comment]
update progress
1. after fix merge.go memory leak Risk, there still cost 3GB in newDecompressToObject
![image](https://user-images.githubusercontent.com/3927687/225316212-bea39020-6791-4d83-90f6-ca4c29f73030.png)


2.  task service would re-scheduler finished (sys_async_task.task_status = 2) task.
- ![企业微信截图_654933f6-8b87-4cb6-b37a-ca7c3e931213](https://user-images.githubusercontent.com/3927687/225317720-7e9c3566-0b4c-4655-a8c4-e0485bca5b94.png)
- ![企业微信截图_8d836fff-c0bc-4751-b1c0-5e9c39823b17](https://user-images.githubusercontent.com/3927687/225317740-e6cd6948-14aa-45cf-a076-a768b551b336.png)

**Comment by xzxiong** [comment]
6 GB is the cach of fileservice cache size.

**Comment by xzxiong** [comment]
在 objectio.newDecompressToObject 被缓存 4790 MB 时，整个进程的内存占用已近 11GB
![image](https://user-images.githubusercontent.com/3927687/225562989-80170460-4733-42f9-94d0-c862ae8cb7aa.png)

进程RSS 占用 11558152 KB ~= 11. GB
![image](https://user-images.githubusercontent.com/3927687/225562787-5f22678f-cee6-4a70-bc45-7242642c032a.png)

**Comment by xzxiong** [comment]
status:
- [ ] (fixed, wait new pr)  bug: free Reader's vector in mpool
- [ ] (fixing, by ziran) CN will re-schduler finished async task in table `mo_task`.`sys_async_task`
- [ ] (try to do NoCache) ETL Merge do cache not welcome data.

**Comment by xzxiong** [comment]
> status:
> 
> * [x]  (fixed, wait new pr)  bug: free Reader's vector in mpool
> * [x]  (fixing, by ziran) CN will re-schduler finished async task in table `mo_task`.`sys_async_task`
> * [x]  (try to do NoCache) ETL Merge do cache not welcome data.

abstract
-  (fixed by #8541)  bug: free Reader's vector in mpool
-  (fixed by #8492) CN will re-schduler finished async task in table `mo_task`.`sys_async_task`
-  (fixed by #8541) ETL Merge do cache not welcome data.

remiain
- [Enhancement]: mpool needs to detect memory incorrectly released #8563
